RecFCCeeCalorimeter/src/components/CaloTowerToolFCCee.cpp

```diff
@@ -37,69 +43,70 @@ StatusCode CaloTowerToolFCCee::initialize() {
     return StatusCode::FAILURE;
   }
 
-  // check if readouts exist & retrieve Module-Theta segmentations
-  // if readout does not exist, reconstruction without this calorimeter part will be performed
-  std::pair<dd4hep::DDSegmentation::Segmentation*, SegmentationType> tmpPair;
+    std::vector<double> listPhiMax;
+  std::vector<double> listThetaMax;
+  listPhiMax.reserve(7);
+  listThetaMax.reserve(7);
 
-  info() << "Retrieving Ecal barrel segmentation" << endmsg;
-  tmpPair = retrieveSegmentation(m_ecalBarrelReadoutName);
-  m_ecalBarrelSegmentation = tmpPair.first;
-  m_ecalBarrelSegmentationType = tmpPair.second;
-  if (tmpPair.first != nullptr && tmpPair.second == SegmentationType::kWrong) {
-    error() << "Wrong type of segmentation" << endmsg;
-    return StatusCode::FAILURE;
-  }
-  if (m_useHalfTower) {
-    m_decoder = m_geoSvc->getDetector()->readout(m_ecalBarrelReadoutName).idSpec().decoder();
+  std::pair<double, double> tmpPair;
+  tmpPair = retrievePhiThetaExtrema(m_ecalBarrelReadoutName);
```

Comment:
I would remove all the m_XXXSegmentationOK booleans and make the job fail with a return StatusCode::FAILURE; If the segmentation is not OK.
We can also do it in a follow-up PR